addons/website/tests/test_res_users.py

```diff
@@ -110,3 +112,19 @@ def test_archive_company_linked_to_website(self):
         # The company cannot be archived because it has a website linked to it
         with self.assertRaises(ValidationError):
             company.action_archive()
+
+    def test_user_become_internal(self):
+        # This tests if the website_id is correctly removed when a user becomes
+        # an internal user.
+        website = self.env['website'].create({'name': "Awesome Website"})
+        # Permit uninvited signup for portal users.
+        website.auth_signup_uninvited = 'b2c'
+        # Disable cross-website for portal users.
+        website.specific_user_account = True
```

Comment:
One sql/orm operation, as we would do in real business code
```suggestion  
        website.write({
            # Permit uninvited signup for portal users.
            'auth_signup_uninvited': 'b2c',
            # Disable cross-website for portal users.
            'specific_user_account': True,
        })
```